File: 2928_distribute_candies_among_children_i/Solution.cs
// LeetCode 2928 - Distribute Candies Among Children I
// https://leetcode.com/problems/distribute-candies-among-children-i/

public class Solution {
    public int DistributeCandies(int n, int limit) {
        int ans = 0;
        for (int i = 0; i <= limit; i++)
            for (int j = 0; j <= limit; j++) {
                int k = n - i - j;
                if (k >= 0 && k <= limit) ans++;
            }
        return ans;
    }
}

File: 2928_distribute_candies_among_children_i/solution.rs
// LeetCode 2928 - Distribute Candies Among Children I
// https://leetcode.com/problems/distribute-candies-among-children-i/

impl Solution {
    pub fn distribute_candies(n: i32, limit: i32) -> i32 {
        let mut ans = 0;
        for i in 0..=limit {
            for j in 0..=limit {
                let k = n - i - j;
                if k >= 0 && k <= limit {
                    ans += 1;
                }
            }
        }
        ans
    }
}
